src/Tribe/Aggregator/Record/Abstract.php

```diff
@@ -1128,59 +1138,131 @@ public function insert_posts( $items = array() ) {
 				$event['post_parent'] = $k;
 			}
 
-			//if we should create a venue or use existing
+			// if we should create a venue or use existing
 			if ( ! empty( $event['Venue']['Venue'] ) ) {
-				$v_id = array_search( $event['Venue']['Venue'], $found_venues );
-				if ( false !== $v_id ) {
-					$event['EventVenueID'] = $v_id;
-				} elseif ( $venue = get_page_by_title( $event['Venue']['Venue'], 'OBJECT', Tribe__Events__Main::VENUE_POST_TYPE ) ) {
-					$found_venues[ $venue->ID ] = $event['Venue']['Venue'];
-					$event['EventVenueID']      = $venue->ID;
-				} else {
-					$event['Venue']['ShowMap']     = $show_map_setting;
-					$event['Venue']['ShowMapLink'] = $show_map_setting;
-					$event['EventVenueID'] = Tribe__Events__Venue::instance()->create( $event['Venue'], $this->meta['post_status'] );
+				if ( ! empty( $item->venue->global_id ) ) {
+					// Did we find a Post with a matching Global ID in History
+					$venue = Tribe__Events__Aggregator__Event::get_post_by_meta( 'global_id_lineage', $item->venue->global_id );
+
+					var_dump( $venue );
+
+					// Save the Venue Data for Updating
+					$venue_data = $event['Venue'];
+
+					if ( $venue ) {
+						$event['EventVenueID'] = $venue->ID;
+						$found_venues[ $venue->ID ] = $event['Venue']['Venue'];
+
+						// Here we might need to update the Venue depending on the main GlobalID
+						// @todo Update Venue
+					} else {
+						$venue_id = array_search( $event['Venue']['Venue'], $found_venues );
+						if ( ! $venue_id ) {
+							$venue = get_page_by_title( $event['Venue']['Venue'], 'OBJECT', Tribe__Events__Venue::POSTTYPE );
+
+							if ( $venue ) {
+								$venue_id = $venue->ID;
+								$found_venues[ $venue_id ] = $event['Venue']['Venue'];
+							}
+						}
 
-					// Log this Venue was created
-					$activity->add( 'venue', 'created', $event['EventVenueID'] );
+						// We didn't find any matching Venue for the provided one
+						if ( ! $venue_id ) {
+							$event['Venue']['ShowMap']     = $show_map_setting;
+							$event['Venue']['ShowMapLink'] = $show_map_setting;
+							$event['EventVenueID'] = Tribe__Events__Venue::instance()->create( $event['Venue'], $this->meta['post_status'] );
+
+							$found_venues[ $event['EventVenueID'] ] = $event['Venue']['Venue'];
+
+							// Log this Venue was created
+							$activity->add( 'venue', 'created', $event['EventVenueID'] );
+
+							// Create the Venue Global ID
+							if ( ! empty( $item->venue->global_id ) ) {
+								update_post_meta( $event['EventVenueID'], Tribe__Events__Aggregator__Event::$global_id_key, $item->venue->global_id );
+							}
+
+							// Create the Venue Global ID History
+							if ( ! empty( $item->venue->global_id_lineage ) ) {
+								foreach ( $item->venue->global_id_lineage as $gid ) {
+									add_post_meta( $event['EventVenueID'], Tribe__Events__Aggregator__Event::$global_id_lineage_key, $gid );
+								}
+							}
+						} else {
+							// Maybe update here
+							// @todo Update Venue
```

Comment:
Ditto here - does this need to be implemented or is it a longer term note (and do we need the empty else clause)?